modules/standard/AutoGpu.chpl

```diff
@@ -28,13 +29,68 @@ module AutoGpu {
   use ChplConfig;
   use Errors;
 
+  /*
+   This attribute can be applied to loops and variable declarations to ensure
+   that they are elgibile for GPU execution. Unlike :annotation:`@assertOnGpu`,
+   this attribute has no runtime effect. It only asserts that the code
+   `could` be executed on the GPU, and not that it `will` be executed.
+
+   When applied to loops, this attribute asserts that the loop itself is
+   GPU-eligible. When applied to variables, this attribute asserts that
+   all promoted expressions and loop expressions (e.g., foreach expressions)
+   are GPU-eligible instead.
+
+
+   .. code-block:: chapel
+
+     // loop version
+     @gpu.assertEligible
+     foreach i in 1..128 { /* ... */ }
+
+     // variable version
+     @gpu.assertEligible
+     var A = (foreach i in 1..128 do i*i) + 1;
+  */
+  @chpldoc.attributeSignature("gpu.assertEligible")
   inline proc chpl__gpuAssertEligibleAttr() {
     if CHPL_LOCALE_MODEL == "gpu" then
       __primitive("assert gpu eligible");
   }
 
+  /*
+   This configuration parameter is used to disable warnings that are emitted
+   when :annotation:`@assertOnGpu` is used in a non-GPU compilation. Since :annotation:`@assertOnGpu`'s
+   runtime semantics are to halt execution if it is not on the GPU, it will
+   always halt execution when the program is not compiled for the GPU. This
+   is likely an issue, so the warning is emitted by default. However, in
+   case the user is aware of this and wants to silence the warning, they can
+   set this configuration parameter to ``true``.
+  */
```

Comment:
Not for this PR, but can we use this `param` in lieu of `CHPL_GPU_NO_CPU_MODE_WARNING` in cpu-as-device? Notionally, they serve the same purpose.

Reply:
I seem to recall config params being used for this purpose, at least during the 2.0 stabilization push, but an environment variable is fine with me as well.